plugins/source/docs.go

```diff
@@ -229,6 +234,102 @@ func formatMarkdown(s string) string {
 }
 
 func formatType(v arrow.DataType) string {
+	if arrow.IsNested(v.ID()) {
+		switch v.ID() {
+		case arrow.STRUCT:
+			s := "struct<"
+			for i, f := range v.(*arrow.StructType).Fields() {
+				if i > 0 {
+					s += ","
+				}
+				s += f.Name + ":" + formatType(f.Type)
+			}
+			return s + ">"
+		case arrow.LIST:
+			return "list<" + formatType(v.(*arrow.ListType).Elem()) + ">"
+		case arrow.LARGE_LIST:
+			return "large_list<" + formatType(v.(*arrow.LargeListType).Elem()) + ">"
```

Comment:
you lose the info about elem nullability here (and it might be vital to know if the elem can be nullable or not (think CH destination))